src/main/java/org/apache/sling/servlets/resolver/internal/SlingServletResolver.java

```diff
@@ -460,6 +469,16 @@ private Resource getErrorResource(final SlingHttpServletRequest request) {
         return res;
     }
 
+    /** @return true if the given Resource is hidden by our resourceHidingPredicate */
+    private boolean isHidden(Resource r) {
```

Comment:
I would explicitly mention that this method requires a non-null parameter (and that for this reason null-checks are missing).
```suggestion
    private boolean isHidden(@NotNull Resource r) {
```

Reply:
Good catch, thank you! Changed as suggested.